### PR TITLE
Bugfix: allow new-style method animation to be used in animation groups

### PR DESCRIPTION
**Pull request by behackl**
## Motivation
New-style method animations (using the `.animate` syntax) can't currently be included in `AnimationGroup` or `Succession`.

## Overview / Explanation for Changes

With this change, animations in the list are checked whether they are a `_AnimationBuilder` -- and if so, the corresponding animation is built and added to `self.animations`.

## Oneline Summary of Changes

```
- Bugfix: allow new-style method animation to be used in animation groups (:pr:`PR NUMBER HERE`)
```

## Testing Status

Added a test for `AnimationGroup` and `Succession`, passing locally.

## Acknowledgements
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)

## Reviewer Checklist
- [x] Newly added functions/classes are either private or have a docstring
- [x] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings
- [ ] The oneline summary has been included [in the wiki](https://github.com/ManimCommunity/manim/wiki/Changelog-for-next-release)

### CODE REVIEW

**Review by leotrs — APPROVED** [review]

**Comment by friedkeenan** [comment]
Perhaps there should be a single function to handle this stuff that could be used in `AnimationGroup` and in `Scene`, something like
```py
def handle_animation(anim):
    if isinstance(anim, _AnimationBuilder):
        return anim.build()
    elif isinstance(anim, Animation):
        return anim

    raise TypeError(...)
```

**Comment by behackl** [comment]
dd54e1fc now delegates building of an animation from an animation factory to a dedicated `prepare_animation` function.

**Review by friedkeenan — COMMENTED** [review]
Looks good to me, just have a couple not very important nitpicks

File: manim/animation/composition.py [hunk 1]
```diff
@@ -29,7 +29,9 @@ def __init__(
         lag_ratio: float = 0,
         **kwargs
     ) -> None:
-        self.animations = animations
+        self.animations = []
+        for anim in animations:
+            self.animations.append(prepare_animation(anim))
```
[the remark this answers is not in the document]
Comment (behackl) [reply manim/animation/composition.py]: Agreed.

File: manim/animation/composition.py [hunk 2]
```diff
@@ -3,8 +3,8 @@
 
 import numpy as np
 
-from ..animation.animation import Animation
-from ..mobject.mobject import Group, Mobject
+from ..animation.animation import Animation, prepare_animation
+from ..mobject.mobject import Group, Mobject, _AnimationBuilder
```
[the remark this answers is not in the document]
Comment (behackl) [reply manim/animation/composition.py]: True, as far as I can tell: 17f1cccb

**Review by friedkeenan — APPROVED** [review]

**Pull request merged**